libpod/container_internal_linux.go

```diff
@@ -66,17 +66,55 @@ func (c *Container) prepare() (err error) {
 		defer wg.Done()
 		// Set up network namespace if not already set up
 		if c.config.CreateNetNS && c.state.NetNS == nil && !c.config.PostConfigureNetNS {
-			saveNetworkStatus = true
 			netNS, networkStatus, createNetNSErr = c.runtime.createNetNS(c)
+
+			tmpStateLock.Lock()
```

Comment:
Is there anyway we can avoid a lock in the main path?

Reply:
We only take it after the work is done - so this shouldn't impact execution time in a significant way, we just hold it during the write to the struct

Reply:
sorry, was confusing with a container lock.